src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs

```diff
@@ -1392,126 +1423,121 @@ private class DebuggerCommandArgument
                 }
 
-                string breakpointType = null;
-                foreach (var commandParameter in command.Parameters)
-                {
-                    if (!commandParameter.Name.Equals("Script", StringComparison.OrdinalIgnoreCase) &&
-                        !commandParameter.Name.Equals("Line", StringComparison.OrdinalIgnoreCase) &&
-                        !commandParameter.Name.Equals("Column", StringComparison.OrdinalIgnoreCase) &&
-                        !commandParameter.Name.Equals("Action", StringComparison.OrdinalIgnoreCase) &&
-                        !commandParameter.Name.Equals("Command", StringComparison.OrdinalIgnoreCase) &&
-                        !commandParameter.Name.Equals("Variable", StringComparison.OrdinalIgnoreCase) &&
-                        !commandParameter.Name.Equals("Mode", StringComparison.OrdinalIgnoreCase))
-                    {
-                        throw new PSArgumentException(commandParameter.Name);
-                    }
-
-                    if (commandParameter.Name.Equals("Line", StringComparison.OrdinalIgnoreCase) ||
-                        commandParameter.Name.Equals("Column", StringComparison.OrdinalIgnoreCase))
-                    {
-                        if (breakpointType != null && breakpointType != "Line")
-                        {
-                            throw new PSArgumentException(commandParameter.Name);
-                        }
+                TryGetParameter<int?>(command, "RunspaceId", out int? runspaceId);
 
-                        breakpointType = "Line";
-                    }
-                    else if (commandParameter.Name.Equals("Command", StringComparison.OrdinalIgnoreCase))
-                    {
-                        if (breakpointType != null)
-                        {
-                            throw new PSArgumentException(commandParameter.Name);
-                        }
+                commands.Clear();
 
-                        breakpointType = "Command";
-                    }
-                    else if (commandParameter.Name.Equals("Variable", StringComparison.OrdinalIgnoreCase))
+                // Any collection comes through remoting as an ArrayList of Objects so we convert each object
+                // into a breakpoint and add it to the list.
+                var bps = new List<Breakpoint>();
+                if (breakpoints != null)
+                {
+                    foreach (object obj in breakpoints)
                     {
-                        if (breakpointType != null)
+                        if (!LanguagePrimitives.TryConvertTo<Breakpoint>(obj, out Breakpoint bp))
                         {
-                            throw new PSArgumentException(commandParameter.Name);
+                            throw new PSArgumentException(DebuggerStrings.BreakpointListContainedANonBreakpoint);
```

Comment:
Should this be non-terminating?

Reply:
I don't think so. Honestly this shouldn't ever happen because the SetBreakpoints() method only takes in an `IEnumerable<Breakpoint>`.

Reply:
but if some one were to directly send a `__Set-Breakpoint` remoting message, they could hypothetically pass any value... but who would do that?